### PR TITLE
Second Commit

### PR DESCRIPTION
**Pull request by StergiosAnastasiadis**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,12 +1,70 @@
...
     <div className="App">
-      <InputField />
+      <Header moviesInsideCart={moviesInsideCart} />
+      <Input setUserSearchInput={setUserSearchInput} data={data} setData={setData}/>
+      <div className="testFlexBox">
+      {
+      (data.length === 0) ? <NoMoviesFound />
+      :(isLoading) ? <Loading />
+      :<div className="movie-cards-container">
+      {data.map((item) => (<Card key={item.id} id={item.id} title={item.title} poster_path={item.poster_path} vote_average={item.vote_average} buttonDisable={moviesInsideCart.some(mic => mic.id === item.id)} addMovieButton={() => addMovieButton({...item})} />))}
```
Comment (gianevag) [inline src/App.js]: You could merge the Card properties {id, title, poster_path, vote_average} to an object!

File: src/App.js [hunk 2]
```diff
@@ -1,12 +1,70 @@
...
+const App = () => {
+
+  const [userSearchInput, setUserSearchInput] = useState("");
+  const [data, setData] = useState(defaultMovieCards);
+  const [isLoading, setIsLoading] = useState(false);
+  const [moviesInsideCart, setMoviesInsideCart] = useState([]);
+  const [isBuyButtonClicked, setIsButtonClicked] = useState(false);
+
+  useEffect(() => {
+    userSearchInput && getData(userSearchInput, setIsLoading)
```
Comment (gianevag) [inline src/App.js]: Good Job,  to create a separate file for API calls!

File: src/App.js [hunk 3]
```diff
@@ -1,12 +1,70 @@
...
+      {
+      (data.length === 0) ? <NoMoviesFound />
+      :(isLoading) ? <Loading />
+      :<div className="movie-cards-container">
+      {data.map((item) => (<Card key={item.id} id={item.id} title={item.title} poster_path={item.poster_path} vote_average={item.vote_average} buttonDisable={moviesInsideCart.some(mic => mic.id === item.id)} addMovieButton={() => addMovieButton({...item})} />))}
+      </div>
+      }
+      <div className="movie-cart">
+        <h5 className="card-title">Buy Movies</h5>
+        {moviesInsideCart && moviesInsideCart.map((item) => (<Cart key={item.id} id={item.id} title={item.title} setMoviesInsideCart={setMoviesInsideCart} />))}
```
Comment (gianevag) [inline src/App.js]: merge id & title props

File: src/components/Card.jsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+
+const Card = ({ id, title, poster_path, buttonDisable, addMovieButton }) => {
+
+    const imgUrl = "https://image.tmdb.org/t/p/original/";
```
Comment (gianevag) [inline src/components/Card.jsx]: you can import the imgUrl from a file that handles the constant strings!

File: src/components/Input.jsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+const Input = ({ setUserSearchInput, data, setData }) => {
+
+  const handleUserSearchInput = (event) => {
+    setUserSearchInput(event.target.value);
+  }
+
+  const lowestToHighest = () => {
+    let newArray = data.sort((a, b) =>  parseFloat(a.vote_average) - parseFloat(b.vote_average));
+    setData([...newArray]);
```
Comment (gianevag) [inline src/components/Input.jsx]: the spread operator is not necessary, same line 16!

File: src/App.js [hunk 6]
```diff
@@ -1,12 +1,70 @@
...
+import Cart from "./components/Cart";
+import getData from "./getData";
+import purchaseMovies from "./purchaseMovies";
 import "./App.css";
+import Loading from "./components/Loading";
+
+const App = () => {
+
+  const [userSearchInput, setUserSearchInput] = useState("");
+  const [data, setData] = useState(defaultMovieCards);
```
Comment (sentonia) [inline src/App.js]: We don't need to initiate dummy movies, we need to be an empty array as initial value and set the movies that you fetch when the app is up.

File: src/App.js [hunk 7]
```diff
@@ -1,12 +1,70 @@
...
+    alert("Successfully Purchased")
+    console.log(res.config.data);
+  }
+
+  const buyMoviesButton = async () => {
+    if(moviesInsideCart.length === 0 || isBuyButtonClicked) { return;}
+    setIsButtonClicked(true);
+    await purchaseMovies(moviesInsideCart).then(res => {
+      res.data.success ? purchaseSuccess(res) : alert("Something went wrong. Please Try Again");
+    })
```
Comment (sentonia) [inline src/App.js]: Try it with react-toastify, it will be more user-friendly!

File: src/components/Card.jsx [hunk 8]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import { imgUrl } from "../urlStrings";
+
+const Card = ({ id, title, poster_path, buttonDisable, addMovieButton }) => {
+    
+    return (
+        <div>
+            <div className="card" id="movie-cards">
+                <button id="add-movie-button" className="btn btn-secondary btn-sm" onClick={addMovieButton} value={id} disabled={buttonDisable}>Add</button>
```
Comment (sentonia) [inline src/components/Card.jsx]: We also need the add button to be transformed to a remove button. When the user adds a movie into the cart, the add button on the movie card should be a remove button and when the user clicks on it, the movie should be removed from the cart.

File: src/components/Cart.jsx [hunk 9]
```diff
@@ -0,0 +1,20 @@
+import React from "react";
+
+const Cart = ({ id, title, setMoviesInsideCart }) => {
+
+    const deleteMovie = () => {
+        setMoviesInsideCart((moviesInsideCart) => {
+            const newArray = [...moviesInsideCart].filter(movie => !(movie.id === id));
```
Comment (sentonia) [inline src/components/Cart.jsx]: why spread operator? The filter() method creates a new array.

File: src/defaultMovieCards.js [hunk 10]
```diff
@@ -0,0 +1,34 @@
+const defaultMovieCards = [
```
Comment (sentonia) [inline src/defaultMovieCards.js]: You don't need this anymore! :)

File: src/purchaseMovies.js [hunk 11]
```diff
@@ -0,0 +1,20 @@
+import axios from "axios";
+
+const purchaseMovies = async (moviesInsideCart) => {
+
+    const url = "https://api.mocklets.com/mock68075/";
+    const config = { headers: {"X-Mocklets-PublicKey" : "txmovies", "X-Mocklets-Checksum" : "830c7cd4a70be6540a4898441ca02951"}};
+    let moviesToBuy ={data:{movies:[]}};
+
+    moviesInsideCart.map((movie) => {
```
Comment (sentonia) [inline src/purchaseMovies.js]: you don't need a map, just use a forEach. 
https://codeburst.io/javascript-map-vs-foreach-f38111822c0f

moviesInsideCart.forEach((movie) => {
        moviesToBuy.data.movies.push(movie.id);
    })

**Pull request merged**